File: rpg/schemas/skills.py
"""schemas.skills — Skill 导入与运行路由请求模型。"""
from __future__ import annotations

from typing import Any

from schemas._common import _BaseRequest


class SkillsImportRequest(_BaseRequest):
    file: Any | None = None


class SkillRunRequest(_BaseRequest):
    cmd: list[Any] | None = None
    command: list[Any] | None = None
    stdin: str | None = None
    timeout_sec: int | None = None

File: rust/crates/rpg-schemas/src/skills.rs
//! schemas.skills — Skill 导入与运行路由请求模型。
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct SkillsImportRequest {
    pub file: Option<Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct SkillRunRequest {
    pub cmd: Option<Vec<Value>>,
    pub command: Option<Vec<Value>>,
    pub stdin: Option<String>,
    pub timeout_sec: Option<i64>,
}
